library/bignum_core.c

```diff
@@ -540,4 +540,18 @@ int mbedtls_mpi_core_get_mont_r2_unsafe( mbedtls_mpi *X,
     return( ret );
 }
 
+void mbedtls_mpi_core_ct_uint_table_lookup( mbedtls_mpi_uint *dest,
+                                        const mbedtls_mpi_uint *table,
+                                        size_t limbs,
+                                        size_t count,
+                                        size_t index )
+{
+    for( size_t i = 0; i < count; i++ )
+    {
+        unsigned char assign = mbedtls_ct_size_bool_eq( i, index );
+        const mbedtls_mpi_uint *current = table + i * limbs;
+        mbedtls_mpi_core_cond_assign( dest, current, limbs, assign );
+    }
+}
```

Comment:
What constant-time functions should go in `constant_time.c` vs `bignum_core.c`?

Reply:
Eventually all functions in `bignum_core` will be constant time. For now constant time functions go to `constant_time.c` by default. In this case, this function will be static and only called from the exponentiation function. It is not marked as static, because the compiler complains about it not being used, will add the qualifier later.